Namco/pacman.py

```diff
@@ -1,42 +1,54 @@
-import numpy as np
-
-
-class PacMan:
-    def __init__(self, map_size=10, max_time_step=100):
-        # PacMan variables
-        self.map_size = map_size
-        self.time_step = 0
-        self.max_time_step = max_time_step
-        self.map = np.zeros((map_size, map_size))
-        self.x = np.random.randint(0, self.map_size)
-        self.y = np.random.randint(0, self.map_size)
-
-    def reset(self):
-        pass
-
-    def step(self):
-        done = False
-
-        # Increasing the timestep
-        self.time_step += 1
-
-        # Terminating after max time steps
-        if self.time_step > self.max_time_step:
-            done = True
-
-        return done
-
-    def render(self):
-        pass
-
-
-if __name__ == "__main__":
-    # Instantiating the environment
-    env = PacMan(map_size=10,
-                 max_time_step=100)
-
-    done_ = False
-    while not done_:
-        done_ = env.step()
-
-
+import numpy as np
+from pytimedinput import timedInput
+
+class PacMan:
+    def __init__(self, map_size=10, max_time_step=100):
+        # PacMan variables
+        self.map_size = map_size
+        self.time_step = 0
+        self.max_time_step = max_time_step
+        self.map = np.zeros((map_size, map_size))
+        self.x = np.random.randint(0, self.map_size)
+        self.y = np.random.randint(0, self.map_size)
+
+    def reset(self):
+        pass
+
+    def step(self, action):
+        done = False
+
+        # Increasing the timestep
+        self.time_step += 1
+
+        # Terminating after max time steps
+        if self.time_step > self.max_time_step:
+            done = True
+
+        return done
+
+    def render(self):
+        pass
+
+    # Automatic stepping if no input (direction) given
+    def auto_step(self, prev_action):

```

Comment:
Ez lehetne select_action() függvény, ami vagy visszaadja az előző akciót, vagy beolvassa a az inputot és visszaad egy action-t. Majd ezt az actiont adjuk be az env.step-nek valahogy így:
action = select_action()
state, reward, done = step(action)
Majd ez alapján az action alapján lépteti pacman-t a step függvény